Photochemistry/src/ReactionNetwork.jl

```diff
@@ -8,87 +8,100 @@
     =#
-    
+
     GV = values(globvars)
-    required = [:all_species, :dz, :ion_species, :num_layers, :reaction_network, :results_dir, :sim_folder_name, :Tn, :Ti, :Te]
+    required = [:all_species, :dz, :ion_species, :num_layers, :reaction_network,
+                :results_dir, :sim_folder_name, :Tn, :Ti, :Te]
     check_requirements(keys(GV), required)
-    
+
     flush(stdout)
     println("Writing out column rates to the reaction log...")
     flush(stdout)
-    
-    # Get the Jrates, needed later
-    Jratedict = Dict([j=>atm_state[j] for j in keys(atm_state) if occursin("J", string(j))]);
-    
-    Mtot = n_tot(atm_state; GV.all_species);
-    
-    # Open the active reactions file
-    the_spreadsheet_file = GV.results_dir*GV.sim_folder_name*"/"*rxn_filename
+
+    # Jrates needed for photochemical reactions
+    Jratedict = Dict([j => atm_state[j] for j in keys(atm_state) if occursin("J", string(j))])
+
+    # Loop through sheets in the spreadsheet
+    the_spreadsheet_file = GV.results_dir * GV.sim_folder_name * "/" * rxn_filename
     active_rxns_original = XLSX.readxlsx(the_spreadsheet_file)
     original_sheets = XLSX.sheetnames(active_rxns_original)
-    
-    # Loop through the sheets
-    for (j, sheet) in enumerate(original_sheets)
 
-        if occursin("Unused", sheet)
+    for sheet in original_sheets
+        if occursin("Unused", sheet) || occursin("Sheet1", sheet)
             continue
-        elseif occursin("Sheet1", sheet)
-            continue
-        else
-            flush(stdout)
-            println("Working on sheet $(sheet)")
-            flush(stdout)
-            df = DataFrame(XLSX.readtable(the_spreadsheet_file, sheet));
+        end
 
-            # Create a new excel column for column rate at the next column
-            insertcols!(df, size(df)[2]+1, :ColumnRate=>[0. for i in collect(1:size(df)[1])])
+        flush(stdout)
+        println("Working on sheet $(sheet)")
+        flush(stdout)
 
-            # Loop through the reactions in each sheet
-            for row in eachrow(df)
-                # collect the reactant names from the sheet;
-                rcols, pcols = get_product_and_reactant_cols(df)
+        df = DataFrame(XLSX.readtable(the_spreadsheet_file, sheet))
 
-                these_reactants = [Symbol(row.:($r)) for r in rcols]
-                these_products = [Symbol(row.:($p)) for p in pcols]
+        # Add new columns for each horizontal column separately
+        for ihoriz in 1:n_horiz
+            insertcols!(df, size(df)[2]+1, Symbol("ColumnRate_$(ihoriz)") => zeros(Float64, size(df)[1]))
+        end
+
+        # Loop through reactions
+        for (row_index, row) in enumerate(eachrow(df))
+            rcols, pcols = get_product_and_reactant_cols(df)
 
-                filter!(x->x!=:none, these_reactants)
-                filter!(x->x!=:none, these_products)
+            these_reactants = [Symbol(row[r]) for r in rcols if row[r] != "none"]
```

Comment:
Similarly here, can you please verify if this makes the intended symbol? I wrote the original version (lines 56 and 57 at left) in an extremely specific way because it was very difficult to get to work... but it's possible new Julia has sorted it, too